### PR TITLE
service/dap: Add launch request support for stackTraceDepth

### PR DESCRIPTION
**Pull request by polinasok**

### CODE REVIEW

File: service/dap/server_test.go [hunk 1]
```diff
@@ -345,135 +345,163 @@ func TestSetBreakpoint(t *testing.T) {
 	})
 }
 
+var expectFrames = func(t *testing.T, stResp *dap.StackTraceResponse, wantStart int, wantFrames int, wantTotal int, wantLine int) {
```
Comment (eliben) [inline service/dap/server_test.go]:
```
func expectFrames(t *testing.T, ...)
```

And please add a comment for this function

File: service/dap/server_test.go [hunk 2]
```diff
@@ -345,135 +345,163 @@ func TestSetBreakpoint(t *testing.T) {
...
+	}
+	if len(stResp.Body.StackFrames) != wantFrames {
+		t.Errorf("\ngot  len(StackFrames)=%d\nwant %d", len(stResp.Body.StackFrames), wantFrames)
+	} else {
+		for i := 0; i < wantFrames; i++ {
+			if stResp.Body.StackFrames[i].Id != wantStart+i {
+				t.Errorf("\ngot  %#v\nwant Id=%d", stResp.Body.StackFrames[i], wantStart+i)
+			}
+		}
+		if wantFrames > 0 && wantLine > 0 && stResp.Body.StackFrames[0].Line != wantLine {
```
Comment (eliben) [inline service/dap/server_test.go]: Should this be `stResp.Body.StackFrames[i]`?

If `[0]` is intended, please add a comment and move it out of the loop

File: service/dap/server_test.go [hunk 3]
```diff
@@ -345,135 +345,163 @@ func TestSetBreakpoint(t *testing.T) {
...
 	})
 }
 
-// runDebugSesion is a helper for executing the standard init and shutdown
+// runDebugSesionWithBPs is a helper for executing the standard init and shutdown
 // sequences for a program that does not stop on entry
-// while specifying unique launch criteria via parameters.
-func runDebugSession(t *testing.T, client *daptest.Client, launchRequest func()) {
+// while specifying breakpoints and unique launch criteria via parameters.
+func runDebugSessionWithBPs(t *testing.T, client *daptest.Client, launchRequest func(), source string, breakpoints []int, onBreakpoints []func()) {
```
Comment (eliben) [inline service/dap/server_test.go:484]: Please document the meaning of the `launchRequest` and `onBreakpoints` args. What is the called function(s) expected to do?

File: service/dap/server_test.go (hunk 1 above)
Comment (polinasok) [reply service/dap/server_test.go]: Done.

File: service/dap/server_test.go (hunk 2 above)
Comment (polinasok) [reply service/dap/server_test.go]: It is out of the loop. It is part of else {}.
Added a comment.

File: service/dap/server_test.go (hunk 3 above)
Comment (polinasok) [reply service/dap/server_test.go:484]: Done

**Review by polinasok — COMMENTED** [review]
PTAL

File: service/dap/server_test.go (hunk 2 above)
Comment (eliben) [reply service/dap/server_test.go]: Oops, my bad.

**Review by eliben — APPROVED** [review]

**Pull request closed without merging**